hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/sequences/AlignedReadPair.java

```diff
@@ -0,0 +1,95 @@
+package org.molgenis.hadoop.pipeline.application.sequences;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import htsjdk.samtools.SAMRecord;
+
+/**
+ * An aligned read pair. Stores the {@link SAMRecord}{@code s} belonging both aligned reads.
+ */
+public class AlignedReadPair
+{
+	/**
+	 * The first read from a read pair.
+	 */
+	private AlignedRead first;
+
+	/**
+	 * The second read from a read pair.
+	 */
+	private AlignedRead second;
+
+	/**
+	 * The read pair type.
+	 */
+	private AlignedReadPairType type;
+
+	public AlignedRead getFirst()
+	{
+		return first;
+	}
+
+	public AlignedRead getSecond()
+	{
+		return second;
+	}
+
+	public AlignedReadPairType getType()
+	{
+		return type;
+	}
+
+	/**
+	 * Generate a new {@link AlignedReadPair} based on a single {@link SAMRecord} {@link List} containing records from
+	 * both reads.
+	 * 
+	 * @param records
+	 *            {@link List}{@code <}{@link SAMRecord}{@code >}
+	 */
+	public AlignedReadPair(List<SAMRecord> records)
+	{
+		divideRecords(records);
+		type = AlignedReadPairType.determineType(first.getType(), second.getType());
+	}
+
+	/**
+	 * Orders and evaluates whether the stored {@link SAMRecord}{@code s} are correct. Note that if the stored
+	 * {@link SAMRecord} {@link List}{@code s} of the {@link AlignedRead}{@code s} or the {@link SAMRecord}{@code s} in
+	 * it are manipulated outside of this class, this method NEEDS TO BE CALLED BEFORE DOING ANYTHING ELSE!
+	 */
```

Comment:
The `AlignedReadPair.update()` method only needs to be called if an instance of this class is created and **afterwards** one of the `AlignedRead`s is adjusted (as this could cause the `AlignedReadPairType` to represent how the data **was** instead of how it **is**). It is currently not used, but still implemented as the `AlignedReadPairType` completely depends on both stored `AlignedRead`s.

The individual `AlignedRead()`s do have their own `update()` already called [in their own constructor](https://github.com/molgenis/hadoop-pipeline/pull/8/files#diff-5a28632cd6f4c7d844c3e43df832d1fdR92). Calling `AlignedReadPair.update()` in the constructor of `AlignedReadPair` would mean the `AlignedRead.update()` would be done twice during initialization of an `AlignedReadPair`.

I'll make some adjustments however to make this clearer.